src/xrc/xh_infobar.cpp

```diff
@@ -0,0 +1,111 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/xrc/xh_infobar.cpp
+// Purpose:     XML resource handler for wxInfoBar
+// Author:      Ilya Sinitsyn
+// Created:     2019/09/25
+// Copyright:   (c) 2019 TT-Solutions SARL
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#if wxUSE_XRC && wxUSE_INFOBAR
+
+#include "wx/xrc/xh_infobar.h"
+
+#include "wx/infobar.h"
+#include "wx/xml/xml.h"
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxInfoBarXmlHandler, wxXmlResourceHandler);
+
+#define XRC_ADD_SHOW_EFFECT(style) m_effectNames[style] = #style;
+
+wxInfoBarXmlHandler::wxInfoBarXmlHandler()
+    : wxXmlResourceHandler(), m_insideBar(false)
+{
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_NONE);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_ROLL_TO_LEFT);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_ROLL_TO_RIGHT);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_ROLL_TO_TOP);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_ROLL_TO_BOTTOM);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_LEFT);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_RIGHT);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_TOP);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_BOTTOM);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_BLEND);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_EXPAND);
+}
+
+wxObject *wxInfoBarXmlHandler::DoCreateResource()
+{
+    if( m_class == wxT("wxInfoBar") )
```

Comment:
```suggestion
    if ( m_class == "wxInfoBar" )
```
(alternatively, use `wxS()`, if you really want to, but avoid `wxT()` in any new code)